spimdisasm/mips/symbols/MipsSymbolBase.py

```diff
@@ -490,7 +499,14 @@ def getNthWordAsString(self, i: int) -> tuple[str, int]:
             return "", -1
 
         skip = rawStringSize // 4
-        comment = self.generateAsmLineComment(localOffset)
+        if rawStringSize == 0:
+            words = [0]
+        elif skip > 0 and self.words[skip - 1] == 0:
+            words = self.words[:skip]
+        else:
+            words = self.words[:skip + 1]
```

Comment:
This is weird. Surely you just want something like `self.words[i:skip]`. What kind of issue where you getting?

Reply:
It leaves out the zero bytes if skip is the end index. If there's a gap of 0x14 bytes between some data, spimdism only outputs 0x10 bytes for example. Null strings have no data also.

```
.align 2
dlabel D_8007C080
    /* 07CC80 8007C080 A5D4A1BCA5D4A1BCA5DEA5C3A5AFA5B9 */ .asciz "ピーピーマックス"
.align 2
.align 2
    /* 07CC94 8007C094  */ .asciz ""
.align 2
.align 2
    /* 07CC98 8007C098  */ .asciz ""
.align 2
.align 2
    /* 07CC9C 8007C09C  */ .asciz ""
.align 2
```

Maybe it's desirable to not include the zeros in the normal strings, but padding bytes are usually shown in other data so we can see them all.

Reply:
Oh, I see. Yes, ideally we want every byte, including the zero-padding.

Either way, the logic of this block is fishy. I'll need to throw more test cases at it